### DISCUSSION TITLE
Calculating ip distance with binary vectors.

### DISCUSSION THREAD

**Discussion by gaozh424** [milvus-io/milvus#26512] [Q&A and General discussion]
In my scenario, I have a large number of image feature vectors with dimensions of 256. Each dimension is represented by an int8 value. I need to calculate the inner product (IP) distance between vectors. While FloatVector can be used for this purpose, it would result in memory usage that is more than four times the actual data size. On the other hand, using BinaryVector does not allow obtaining the IP distance. 
As I have limited memory, do you have any suggestions or recommendations?

**Comment by yhmo** [comment]
Currently, Milvus only supports two types of vectors:
1. float32 values, each dimension is a float32
2. bytes, each dimension is a bit, each byte is 8 dimensions

You can choose IVF_SQ8 to reduce memory usage. The index size of IVF_SQ8 is only about 25% ~ 30% of the original data size. Assume you have 1GB 256d vectors(each dim is an int8), after the data is inserted into Milvus, the data size is 4GB(each dim is a float32). After the index is finished, the index size is 1GB ~ 1.2GB(almost equal to your original data size).

**Reply by gaozh424** [reply]
Thanks! I will try it.
